Source/NSView.m

```diff
@@ -5242,6 +5273,145 @@ - (void) _layoutEngineDidChangeAlignmentRect
 
 @end
 
+@implementation NSView (NSConstraintBasedLayoutCoreMethods)
+
+- (void) updateConstraintsForSubtreeIfNeeded
+{
+  NSArray *subviews = [self subviews];
+  FOR_IN (NSView *, subview, subviews)
+    [subview updateConstraintsForSubtreeIfNeeded];
+  END_FOR_IN (subviews);
+
+  if ([self needsUpdateConstraints])
+    {
+      [self updateConstraints];
+    }
+}
+
+- (void) updateConstraints
+{
+  if ([self translatesAutoresizingMaskIntoConstraints] &&
+      [self superview] != nil)
+    {
+      NSArray *autoresizingConstraints = [NSAutoresizingMaskLayoutConstraint
+          constraintsWithAutoresizingMask: [self autoresizingMask]
+                                  subitem: self
+                                    frame: [self frame]
+                                superitem: [self superview]
+                                   bounds: [[self superview] bounds]];
+      [self addConstraints: autoresizingConstraints];
+    }
+
+  [self _setNeedsUpdateConstraints: NO];
+}
+
+- (NSLayoutPriority)contentCompressionResistancePriorityForOrientation:(NSLayoutConstraintOrientation)orientation {
+  if (orientation == NSLayoutConstraintOrientationHorizontal) {
+    return _compressionPriorities.horizontal;
+  } else {
+    return _compressionPriorities.vertical;
+  }
+}
+
+- (void)setContentCompressionResistancePriority:(NSLayoutPriority)priority forOrientation:(NSLayoutConstraintOrientation)orientation {
+    if (orientation == NSLayoutConstraintOrientationHorizontal) {
+      _compressionPriorities.horizontal = priority;
+    } else {
+      _compressionPriorities.vertical = priority;
+    }
+}
+
+- (NSLayoutPriority)contentHuggingPriorityForOrientation:(NSLayoutConstraintOrientation)orientation {
+  GSIntrinsicContentSizePriority priorities = _huggingPriorities;
+  if (orientation == NSLayoutConstraintOrientationHorizontal) {
+    return priorities.horizontal;
+  } else {
+    return priorities.vertical;
+  }
+}
+
+- (void)setContentHuggingPriority:(NSLayoutPriority)priority forOrientation:(NSLayoutConstraintOrientation)orientation
+{
+    if (orientation == NSLayoutConstraintOrientationHorizontal) {
+      _huggingPriorities.horizontal = priority;
+    } else {
+      _huggingPriorities.vertical = priority;
+    }
+}
+
+@end
+
+@implementation NSView (NSConstraintBasedLayoutInstallingConstraints)
+
+- (GSAutoLayoutEngine*) _getOrCreateLayoutEngine
```

Comment:
Maybe move the method `_layoutEngine` to here as well?

Reply:
I have moved this method into the private category. The other methods in this category are all public.